tags-config-service-impl/src/main/java/org/hypertrace/tag/config/service/ConfigServiceCoordinatorImpl.java

```diff
@@ -0,0 +1,122 @@
+package org.hypertrace.tag.config.service;
+
+import static org.hypertrace.tag.config.service.TagConfigConstants.TAG_CONFIG_RESOURCE_NAME;
+import static org.hypertrace.tag.config.service.TagConfigConstants.TAG_CONFIG_RESOURCE_NAMESPACE;
+
+import com.google.protobuf.Value;
+import io.grpc.Channel;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.SneakyThrows;
+import org.hypertrace.config.proto.converter.ConfigProtoConverter;
+import org.hypertrace.config.service.v1.ConfigServiceGrpc;
+import org.hypertrace.config.service.v1.ConfigServiceGrpc.ConfigServiceBlockingStub;
+import org.hypertrace.config.service.v1.DeleteConfigRequest;
+import org.hypertrace.config.service.v1.DeleteConfigResponse;
+import org.hypertrace.config.service.v1.GetAllConfigsRequest;
+import org.hypertrace.config.service.v1.GetAllConfigsResponse;
+import org.hypertrace.config.service.v1.GetConfigRequest;
+import org.hypertrace.config.service.v1.GetConfigResponse;
+import org.hypertrace.config.service.v1.UpsertConfigRequest;
+import org.hypertrace.config.service.v1.UpsertConfigResponse;
+import org.hypertrace.core.grpcutils.client.RequestContextClientCallCredsProviderFactory;
+import org.hypertrace.core.grpcutils.context.RequestContext;
+import org.hypertrace.tag.config.service.v1.Tag;
+
+public final class ConfigServiceCoordinatorImpl implements ConfigServiceCoordinator {
+  private final ConfigServiceBlockingStub configServiceBlockingStub;
+
+  public ConfigServiceCoordinatorImpl(Channel configChannel) {
+    this.configServiceBlockingStub =
+        ConfigServiceGrpc.newBlockingStub(configChannel)
+            .withCallCredentials(
+                RequestContextClientCallCredsProviderFactory.getClientCallCredsProvider().get());
+  }
+
+  @Override
+  public Tag upsertTag(RequestContext requestContext, Tag tag) {
+    try {
+      UpsertConfigRequest upsertConfigRequest =
+          UpsertConfigRequest.newBuilder()
+              .setResourceName(TAG_CONFIG_RESOURCE_NAME)
+              .setResourceNamespace(TAG_CONFIG_RESOURCE_NAMESPACE)
+              .setConfig(convertToGenericFromTag(tag))
+              .setContext(tag.getId())
+              .build();
+      UpsertConfigResponse upsertConfigResponse = upsertConfig(requestContext, upsertConfigRequest);
+      return convertToTagFromGeneric(upsertConfigResponse.getConfig());
+    } catch (Exception e) {
+      throw e;
```

Comment:
If you are just catching and throwing e, then why have try/catch block? Please remove it. That is the default behaviour.